### PR TITLE
DRILL-3340: Add operator metrics registry for metric definitions

### PR DESCRIPTION
**Pull request by sudheeshkatkam**
+ Display metrics as a table within an operator profile panel
+ Rename FragmentStats#getOperatorStats to newOperatorStats

### CODE REVIEW

**Comment by sudheeshkatkam** [comment]
I am not a UI expert; where is the best place to present operator metrics as a table in our query profile page?

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/OperatorWrapper.java [hunk 1]
```diff
@@ -130,4 +136,32 @@ public void addSummary(TableBuilder tb) {
...
+
+      builder.appendCell(
+          new OperatorPathBuilder()
+              .setMajor(major)
+              .setMinor(ip.getRight())
+              .setOperator(op)
+              .build(),
+          null);
+      for (final MetricValue metric : op.getMetricList()) {
+        builder.appendInteger(metric.getLongValue(), null);
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/OperatorWrapper.java]: the code builds the table header from the first profile's metric list, then assumes all minor fragments will have the metrics in the same order of the first profile.

Can you confirm that this assumption is always true ?

**Comment by adeneche** [comment]
I think the way metrics are displayed is fine, and it's definitely an improvement compared to no metrics at all.

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/OperatorWrapper.java (hunk 1 above)
Comment (sudheeshkatkam) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/OperatorWrapper.java]: It's a bad assumption that I made :)

File: exec/java-exec/src/main/java/org/apache/drill/exec/ops/OperatorMetricRegistry.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import org.apache.drill.exec.physical.impl.mergereceiver.MergingRecordBatch;
+import org.apache.drill.exec.physical.impl.partitionsender.PartitionSenderRootExec;
+import org.apache.drill.exec.physical.impl.unorderedreceiver.UnorderedReceiverBatch;
+import org.apache.drill.exec.proto.UserBitShared.CoreOperatorType;
+
+/**
+ * Registry of operator metrics.
+ */
+public class OperatorMetricRegistry {
+//  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(OperatorToMetricsMapping.class);
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/ops/OperatorMetricRegistry.java]: can you change the class passed to `getLogger()` from `OperatorToMetricsMapping` to `OperatorMetricRegistry` ?

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/OperatorWrapper.java [hunk 3]
```diff
@@ -130,4 +136,48 @@ public void addSummary(TableBuilder tb) {
...
+
+      builder.appendCell(
+          new OperatorPathBuilder()
+              .setMajor(major)
+              .setMinor(ip.getRight())
+              .setOperator(op)
+              .build(),
+          null);
+
+      final Map<Integer, Number> valueMap = Maps.newHashMap(); // metric id --> metric value
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/OperatorWrapper.java]: we don't really need a Map here, right ?

Comment (sudheeshkatkam) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/OperatorWrapper.java]: I had already made this change (testing right now).

**Comment by adeneche** [comment]
I have 2 minor comments that need to be addressed, otherwise LGTM
+1

**Pull request closed without merging**